dgisim/src/dices.py

```diff
@@ -162,6 +175,205 @@ def just_satisfy(self, requirement: AbstractDices) -> bool:
+                [(element,
+                  self._how_much_omni_and_first_priority_spends(
+                      element_supply=supply[element],
+                      element_need=need[element],
+                      omni_need=need[Element.OMNI],
+                      element_priority=element in first_priority_elements,
+                      omni_supply=supply[Element.OMNI])
+                  )
+                 for element in _PURE_ELEMS]
+
+            # TODO rewrite minimum get
+            array = list(sorted(array, key=comparision))
+
+            least_spend = array[0][1]
+            least_spend_omni_elements = []
+            if least_spend != BIG_INT:
+                for el in array:
+                    if el[1] == least_spend:
+                        least_spend_omni_elements.append(el[0])
+            else:
+                raise Exception("Cant feed OMNI requirement")
+
+            assert least_spend_omni_elements, "Unknown error"
+
+            precedence_of_least_spend_omni = {element: get_precedence(element)
+                                              for element
+                                              in least_spend_omni_elements}
+
+            max_precedence_omni_potential_fillers: List[Element] = []
+            max_precedence = max(precedence_of_least_spend_omni.values())
+            for key, val in precedence_of_least_spend_omni.items():
+                if val == max_precedence:
+                    max_precedence_omni_potential_fillers.append(key)
+
+            # TODO: выбор наилучшего заполнителя OMNI - now just first
+            filler_element: Element = max_precedence_omni_potential_fillers[0]
+
+            # how many omni and how many filler (we already know Element which
+            # is filler)
+            if supply[filler_element] >= need[Element.OMNI]:
+                filler_element_count = need[Element.OMNI]
+                filler_omni_count = 0
+            else:
+                filler_element_count = supply[filler_element]
+                filler_omni_count = need[Element.OMNI] - filler_element_count
+
+            result_dct[filler_element] += filler_element_count
+            result_dct[Element.OMNI] += filler_omni_count
+
+            need[Element.OMNI] -= filler_element_count + filler_omni_count
+            supply[filler_element] -= filler_element_count
+            supply[Element.OMNI] -= filler_omni_count
+
+            assert need[Element.OMNI] == 0, "need element omni not 0"
+            assert supply[filler_element] >= 0, "supply <0"
+            assert supply[Element.OMNI] >= 0, "supply <0"
+
+        # 2 step - fill the Elements needs and ANY (files ANY parallely)
+
+        global ELEMENTS_BY_DECREASING_GLOBAL_PRIORITY
+
+        reverse_order_need_list: List[Element] = list(
+            sorted(ELEMENTS_BY_DECREASING_GLOBAL_PRIORITY,
+                   key=lambda el: get_precedence(el)
+                   ))[::-1]  # + [Element.ANY]
```

Comment:
instead of reversing the sorted list, adding another O(n) operation, it'd be better if you make the `key=lambda el: -get_precedence(el)`.

Also, `lambda el: -get_precedence(el)` is equivalent to `get_precedence`, so you don't need to wrap it with a _lambda_.

Reply:
done